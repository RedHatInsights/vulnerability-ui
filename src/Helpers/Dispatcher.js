import { addNotification } from '@red-hat-insights/insights-frontend-components/components/Notifications';
import ReducerRegistry from '../Utilities/ReducerRegistry';
export function dispatchAction(actionCreator) {
    const store = ReducerRegistry.getStore();
    return store.dispatch(actionCreator).catch(res => {
        if ('json' in res) {
            let parsedMessage = res.json();
            parsedMessage.then(arr => {
                for (let err of arr.errors) {
                    store.dispatch(
                        addNotification({
                            variant: 'danger',
                            title: 'Error ' + err.status,
                            description: err.detail
                        })
                    );
                }
            });
        } else {
            store.dispatch(
                addNotification({
                    variant: 'danger',
                    title: 'Error ' + res.status,
                    description: res.statusText
                })
            );
        }
    });
}
