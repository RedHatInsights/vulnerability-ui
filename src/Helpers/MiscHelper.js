import { SecurityIcon, UnknownIcon } from '@patternfly/react-icons';
import { SortByDirection } from '@patternfly/react-table';
import findIndex from 'lodash/findIndex';
import propTypes from 'prop-types';
import React from 'react';
import { impactColorList, impactList, PUBLIC_DATE_OPTIONS } from './constants';
import qs from 'query-string';
import { IntlProvider } from '@redhat-cloud-services/frontend-components-translations';

export const dataShape = propTypes.shape({
    data: propTypes.oneOfType([propTypes.object, propTypes.array]),
    meta: propTypes.object,
    isLoading: propTypes.bool
});

/**
 * Based on the  allowedParams it will construct the arguments for the API call
 * which is defined in the vulnerabilities-client. vulnerabilities-client is written is typescript
 * so we need to define and pass every single parameter. Order in allowedParams is important
 *
 * @param {Object} apiProps
 * @param {Array} allowedParams - order should align with the params in the vulnerabilities-client
 */
export function constructParameters(apiProps, allowedParams) {
    if (apiProps) {
        Object.keys(apiProps).forEach(
            key => (
                apiProps[key] === undefined
                || apiProps[key] === ''
            )
            && delete apiProps[key]
        );

        const params = allowedParams.map(item => apiProps[item]);

        return params;
    }

    return [];
}

// TODO DRY:similar to constructParameters
export function constructURLParameters(urlParams, allowedParams) {
    if (urlParams) {
        const params = { ...urlParams };
        Object.keys(urlParams).forEach(
            key => (
                params[key] === undefined
                || params[key] === ''
                || !allowedParams.includes(key)
                || params[key] === false
            )
            && delete params[key]

        );
        return params;
    }
}

export function formatDate(date = new Date(), includeTime = false) {
    const prepend = (number) => `${`${number}`.length === 1 ? '0' : ''}${number}`;
    const toFormat = new Date(date);
    if (toFormat instanceof Date && !isNaN(toFormat)) {
        const year = toFormat.getFullYear();
        const month = toFormat.getMonth() + 1; // month is zero indexed
        const day = toFormat.getDate();
        const hour = toFormat.getUTCHours();
        const minute = toFormat.getUTCMinutes();

        return includeTime
            ? `${year}-${prepend(month)}-${prepend(day)} ${prepend(hour)}:${prepend(minute)} UTC`
            : `${year}-${prepend(month)}-${prepend(day)}`;
    }

    return toFormat;
}

/**
 * Contructs the filters which have more than one parameter that needs to be passed
 * and resets all the parameteres when it's needed
 *
 * @param {Object} filterParams
 * @return {Object}
 */
export function constructFilterParameters(filterParams) {
    if (Object.prototype.hasOwnProperty.call(filterParams, 'publish_date')) {
        let publicEntry = PUBLIC_DATE_OPTIONS.find(item => item.value === filterParams.publish_date);

        if (typeof publicEntry !== 'undefined') {
            filterParams.public_from = publicEntry.from && formatDate(publicEntry.from);
            filterParams.public_to = publicEntry.to && formatDate(publicEntry.to);
        } else {
            filterParams.public_from = undefined;
            filterParams.public_to = undefined;
            filterParams.publish_date = undefined;
        }
    }

    return filterParams;
}

export function getImpactDetails(impact) {
    if (Object.keys(impactList).includes(impact)) {
        return {
            title: impactList[impact].title,
            color: impactList[impact].color,
            text: impactList[impact].text,
            titleContent: (
                <SecurityIcon size="lg" />
            )
        };
    } else {
        return {
            title: 'Unknown',
            color: impactColorList.unknown,
            text: '',
            titleContent: (
                <UnknownIcon size="lg" />
            )
        };
    }
}

export const createSortBy = (header, value) => {
    if (value) {
        let direction = value.indexOf('-') === 0 ? SortByDirection.desc : SortByDirection.asc;
        value = value.replace(/^(-|\+)/, '');
        const index = findIndex(header, item => item.key === value);
        let sort = {
            index,
            direction
        };
        return sort;
    }

    return {};
};

export const handleSortColumn = (key, direction, header, currentSort, apply) => {
    let columnName = header[key].key;
    const useDefault = currentSort && currentSort.substr(1) !== columnName;
    if (direction === SortByDirection.desc || useDefault) {
        columnName = '-' + columnName;
    }

    apply({ sort: columnName });
};

export const updateStateSet = (stateSet, names, value) => {
    [].concat(names).forEach(name => {
        value ? stateSet.add(name) : stateSet.delete(name);
    });
    return stateSet;
};

export const useUrlParams = (allowedParams) => {
    const url = new URL(window.location);
    const urlParams = qs.parse(url.search);

    const setUrlParams = (parameters) => {
        const searchParams = qs.stringify(constructURLParameters(parameters, allowedParams));
        window.history.replaceState(null, null, `${url.origin}${url.pathname}?${searchParams}`);
    };

    return [urlParams, setUrlParams];
};

export const updateRef = (meta, apply) => {
    const pages = parseInt(meta.pages);
    const page = parseInt(meta.page);
    const cvesCount = parseInt(meta.cvesCount);
    const goTo = (pages === page && page > 1 && cvesCount === 1) ? (pages - 1) : page;

    apply({ page: goTo });
};

export const mountWithIntl = (Component) => {
    const wrapper = mount(Component, {
        wrappingComponent: IntlProvider
    });
    const provider = wrapper.getWrappingComponent();
    provider.setProps({ locale: 'en' });

    return wrapper;
};

export const shallowWithIntl = (Component) => {
    const wrapper = shallow(Component, {
        wrappingComponent: IntlProvider
    });
    const provider = wrapper.getWrappingComponent();
    provider.setProps({ locale: 'en' });

    return wrapper;
};

export const insertIf = (condition, ...elements) => {
    return condition ? elements : [];
};
