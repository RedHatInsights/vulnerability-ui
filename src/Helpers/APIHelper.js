let BASE_ROUTE = '/api/vulnerability/v1';
import { GitApi } from '@redhat-cloud-services/vulnerabilities-client';
import { SystemProfileApi } from '@redhat-cloud-services/host-inventory-client';
import instance from '../Utilities/interceptors';
import { constructParameters } from './MiscHelper';

let api = new GitApi(undefined, BASE_ROUTE, instance);

const INVENTORY_API_BASE = '/api/inventory/v1';
const systemProfile = new SystemProfileApi(undefined, INVENTORY_API_BASE, instance);

const defaultParams = [
    'filter',
    'limit',
    'offset',
    'page',
    'page_size',
    'sort'
];

export function getCveDetails(synopsis) {
    const result = api.getCveDetails(synopsis);
    return result;
}

export function getSystemDetails(system) {
    const result = api.getSystemDetails(system);
    return result;
}

export function setSystemCveStatus(params) {
    let result = api.setStatus(params);
    return result;
}

export function optOutSystems(systems, optOut) {
    const data = { opt_out: optOut, inventory_id: new Array(...systems) };
    let result = api.setSystemsOptOut(data);
    return result;
}

export function setBusinessRisk(params) {
    let result = api.setCveBusinessRisk(params);
    return result;
}

export function setCveStatus(params) {
    let result = api.setCveStatus(params);
    return result;
}

/**
 * /vulnerabilities/cves
 * /vulnerabilities/cves/ids
 */
let cveParams = [
    'cvss_from',
    'cvss_to',
    'public_from',
    'public_to',
    'impact',
    'data_format',
    'business_risk_id',
    'status_id',
    'rule_presence',
    'tags',
    'sap_sids',
    'sap_system',
    'known_exploit',
    'affecting',
    'rhel_version'
];

export function getCveListByAccount(apiProps) {
    let parameterNames = [
        ...defaultParams,
        ...cveParams,
        'report',
        'advanced_report',
        'ansible',
        'mssql'
    ];
    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getCveList(...parameterArray);
    return result;
}

export function getCveIdsList(apiProps) {
    let parameterNames = [
        ...defaultParams,
        ...cveParams,
        'ansible',
        'mssql'
    ];

    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getCveIdsList(...parameterArray);
    return result;
}

/**
 * /cves/ID/affected_systems
 */
let systemsByCVEparams = [
    'status_id',
    'data_format',
    'uuid',
    'rule_key',
    'rule_presence',
    'rule',
    'tags',
    'sap_sids',
    'sap_system',
    'show_advisories',
    'advisory',
    'rhel_version',
    'first_reported_from',
    'first_reported_to',
    'advisory_available',
    'remediation'
];

export function getAffectedSystemsByCVE({ id, ...apiProps }) {
    let parameterNames = [
        ...defaultParams,
        ...systemsByCVEparams,
        'report',
        'ansible',
        'mssql'
    ];

    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getAffectedSystemsByCve(id, ...parameterArray);
    return result;
}

export function getAffectedSystemsIdsByCve({ id, ...apiProps }) {
    let parameterNames = [
        ...defaultParams,
        ...systemsByCVEparams,
        'ansible',
        'mssql'
    ];

    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getAffectedSystemsIdsByCve(id, ...parameterArray);
    return result;
}

/**
 * /systems/ID/cves
 * /systems/ID/cves/ids
 */
let cveBySystemParams = [
    'cvss_from',
    'cvss_to',
    'public_from',
    'public_to',
    'impact',
    'status_id',
    'data_format',
    'business_risk_id',
    'rule_presence',
    'show_advisories',
    'advisory',
    'rule_key',
    'known_exploit',
    'first_reported_from',
    'first_reported_to',
    'advisory_available',
    'remediation'
];

export function getCveListBySystem(apiProps) {
    const { system } = apiProps;
    let parameterNames = [
        ...defaultParams,
        ...cveBySystemParams,
        'report'
    ];

    if (apiProps && system) {
        Object.keys(apiProps).forEach(key => (apiProps[key] === undefined || apiProps[key] === '') && delete apiProps[key]);
        const params = parameterNames.map(item => apiProps[item]);
        return api.getCveListBySystem(system, ...params).catch(err => {
            if (err && err.status === '404') {
                return { errors: err };
            }

            throw err;
        });
    }
}

export function getCveIdsBySystem(apiProps) {
    const { system } = apiProps;
    let parameterNames = [
        ...defaultParams,
        ...cveBySystemParams
    ];

    if (apiProps && system) {
        Object.keys(apiProps).forEach(key => (apiProps[key] === undefined || apiProps[key] === '') && delete apiProps[key]);
        const params = parameterNames.map(item => apiProps[item]);
        return api.getCveIdsBySystem(system, ...params).catch(err => {
            if (err && err.status === '404') {
                return { errors: err };
            }

            throw err;
        });
    }
}

/**
 * /systems/ids
 * /systems
 */
let systemsParams = [
    'data_format',
    'stale',
    'uuid',
    'tags',
    'sap_sids',
    'sap_system',
    'excluded',
    'rhel_version'
];

export function getSystems(apiProps) {
    let parameterNames = [
        ...defaultParams,
        ...systemsParams,
        'report',
        'ansible',
        'mssql'
    ];
    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getSystemsList(...parameterArray);
    return result;
}

export function getSystemsIds(apiProps) {
    let parameterNames = [
        ...defaultParams,
        ...systemsParams
    ];
    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getSystemsIds(...parameterArray);
    return result;
}

export function getAnnouncement() {
    let result = api.getAnnouncement();
    return result;
}

export function getDashbar(apiProps) {
    let parameterNames = [
        'tags',
        'sap_sids',
        'sap_system',
        'ansible',
        'mssql'
    ];
    let parameterArray = constructParameters(apiProps, parameterNames);
    let result = api.getDashbar(...parameterArray);
    return result;
}

export function getOperatingSystems() {
    return systemProfile.apiSystemProfileGetOperatingSystem();
}

export function fetchPdfReport(template, filters) {
    const CRC_PDF_GENERATE_API = '/api/crc-pdf-generator/v1/generate';
    const url = new URL(CRC_PDF_GENERATE_API,  window.location.origin);

    return fetch(url, {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },

        body: JSON.stringify({
            service: 'vulnerability',
            template,
            params: {
                ...filters
            }
        })
    }, 50000)
        .then((response) => response.blob());
}
