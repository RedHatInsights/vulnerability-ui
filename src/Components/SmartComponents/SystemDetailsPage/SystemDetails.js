import React from 'react';
import propTypes from 'prop-types';
import { connect } from 'react-redux';
import { ConnectedSystemCves } from '../SystemCves/SystemCves';
import {
    EmptyStateExcludedSystem,
    NotAuthorized
} from '../../PresentationalComponents/EmptyStates/EmptyStates';
import { Spinner } from '@redhat-cloud-services/frontend-components/Spinner';
import { intl } from '../../../Utilities/IntlProvider';
import messages from '../../../Messages';
import { PERMISSIONS } from '../../../Helpers/constants';
import { useRbac } from '../../../Helpers/Hooks';

const SystemDetails = ({ entity, isOptOut, optOutSystemHandler, loaded }) => {
    const [[
        canExport,
        canReadVulnerabilityResults,
        canEditPairStatus,
        canReadExcluded
    ], isRbacLoading] = useRbac([
        PERMISSIONS.basicReporting,
        PERMISSIONS.readVulnerabilityResults,
        PERMISSIONS.setPairStatus,
        PERMISSIONS.readExcluded]
    );

    if (isRbacLoading) {
        return <Spinner centered />;
    } else if (canReadVulnerabilityResults) {
        if (entity && loaded && isOptOut) {
            if (canReadExcluded) {
                return <EmptyStateExcludedSystem buttonAction={() => optOutSystemHandler(entity.display_name, false)} />;
            } else {
                return <NotAuthorized title={intl.formatMessage(messages.notAuthorizedNoAccessForSystem)} />;
            }
        } else {
            return <ConnectedSystemCves
                entity={{ id: entity?.id, display_name: entity?.display_name }}
                showHeaderLabel
                setPageTitle
                canExport={canExport}
                canEditPairStatus={canEditPairStatus}
            />;
        }
    } else {
        return <NotAuthorized />;
    }
};

SystemDetails.propTypes = {
    entity: propTypes.object,
    isOptOut: propTypes.bool,
    loaded: propTypes.bool,
    optOutSystemHandler: propTypes.func
};

function mapStateToProps({ entityDetails: { entity, opt_out: isOptOut, loaded } }) {
    return {
        entity,
        isOptOut,
        loaded
    };
}

export default connect(
    mapStateToProps,
    null
)(SystemDetails);
