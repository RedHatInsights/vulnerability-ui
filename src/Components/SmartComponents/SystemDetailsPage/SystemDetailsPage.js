import { Main } from '@redhat-cloud-services/frontend-components/Main';
import { addNotification } from '@redhat-cloud-services/frontend-components-notifications/redux';
import propTypes from 'prop-types';
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { fetchSystemDetails, optOutSystemAction } from '../../../Store/Actions/Actions';
import { systemDetailsPageStore } from '../../../Store/Reducers/SystemDetailsPageStore';
import ReducerRegistry from '../../../Utilities/ReducerRegistry';
import Header from '../../PresentationalComponents/Header/Header';
import { injectIntl } from 'react-intl';
import { PATHS, ReadOnlyNotification } from '../../../Helpers/constants';
import messages from '../../../Messages';

import { AppInfo, DetailWrapper, InventoryDetailHead } from '@redhat-cloud-services/frontend-components/Inventory';

const InventoryDetail = ({ fetchSystemDetails, systemDetails = {}, optOutSystem, match, addNotification, intl, errors }) => {
    const inventoryId = match.params.inventoryId;

    useEffect(() => {
        fetchSystemDetails(inventoryId);
    }, [fetchSystemDetails, inventoryId]);

    const notifications = systemName => ({
        exclude: {
            success: {
                variant: 'success',
                title: intl.formatMessage(messages.notificationExcludeSuccess, { count: 1, systemName })
            },
            failure: {
                variant: 'danger',
                autoDismiss: false,
                title: intl.formatMessage(messages.notificationExcludeFailureTitle, { count: 1, systemName }),
                description: intl.formatMessage(messages.notificationDetailExcludeFailureBody, { count: 1 })
            }
        },
        include: {
            success: {
                variant: 'success',
                title: intl.formatMessage(messages.notificationIncludeSuccessTitle, { count: 1, systemName }),
                description: intl.formatMessage(messages.notificationIncludeSuccessBody, { count: 1 })
            },
            failure: {
                variant: 'danger',
                autoDismiss: false,
                title: intl.formatMessage(messages.notificationIncludeFailureTitle, { count: 1, systemName }),
                description: intl.formatMessage(messages.notificationDetailIncludeFailureBody, { count: 1 })
            }
        }
    });

    const handleOptOutSystem = toExclude => {
        optOutSystem(inventoryId, toExclude).then(() => {
            const systemName = systemDetails.entity.display_name;

            fetchSystemDetails(inventoryId);
            addNotification(toExclude
                ? notifications(systemName).exclude.success
                : notifications(systemName).include.success);
        }).catch(error => {
            const systemName = systemDetails.entity.display_name;

            addNotification(error.status === '503' ? ReadOnlyNotification : toExclude
                ? notifications(systemName).exclude.failure
                : notifications(systemName).include.failure);
        });
    };

    const { opt_out: isOptOut = false, entity, loaded } = systemDetails;
    return (
        <DetailWrapper>
            <Header
                title=""
                breadcrumbs={[
                    {
                        title: PATHS.systemsPage.title,
                        to: PATHS.systemsPage.to,
                        loaded: true
                    },
                    {
                        title: entity?.display_name || intl.formatMessage(messages.invalidSystem),
                        isActive: true,
                        loaded
                    }
                ]}
            >
                {!errors && (
                    <InventoryDetailHead
                        onLoad={({ mergeWithEntities, mergeWithDetail, INVENTORY_ACTION_TYPES }) => {
                            ReducerRegistry.register({
                                ...mergeWithEntities(),
                                ...mergeWithDetail(systemDetailsPageStore(INVENTORY_ACTION_TYPES, handleOptOutSystem))
                            });
                        }}
                        hideBack
                        actions={
                            entity && [
                                isOptOut
                                    ? {
                                        title: intl.formatMessage(messages.inventoryKebabOptionsResume),
                                        onClick: () => handleOptOutSystem(false)
                                    }
                                    : {
                                        title: intl.formatMessage(messages.inventoryKebabOptionsExclude),
                                        onClick: () => handleOptOutSystem(true)
                                    }
                            ]
                        }
                    />
                )}
            </Header>
            <Main>
                <AppInfo optOutSystemHandler={handleOptOutSystem} />
            </Main>
        </DetailWrapper>
    );
};

InventoryDetail.propTypes = {
    fetchSystemDetails: propTypes.func,
    systemDetails: propTypes.object,
    optOutSystem: propTypes.func,
    match: propTypes.object,
    addNotification: propTypes.func,
    intl: propTypes.any,
    errors: propTypes.object
};

function mapStateToProps(state) {
    return {
        errors: state.SystemCvesStore.cveList.payload.errors,
        systemDetails: state.entityDetails
    };
}

const mapDispatchToProps = dispatch => {
    return {
        optOutSystem: (system, status) => dispatch(optOutSystemAction(system, status)),
        fetchSystemDetails: system => dispatch(fetchSystemDetails(system)),
        addNotification: notification => dispatch(addNotification(notification))
    };
};

export default withRouter(
    injectIntl(
        connect(
            mapStateToProps,
            mapDispatchToProps
        )(InventoryDetail)
    )
);
