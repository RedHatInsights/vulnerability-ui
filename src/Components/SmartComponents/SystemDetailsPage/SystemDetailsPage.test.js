import SystemDetailsPage from './SystemDetailsPage';
import toJson from 'enzyme-to-json';
import { mountWithIntl } from '../../../Helpers/MiscHelper';
import { BrowserRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from "react-redux";
import { InventoryDetailHead } from '@redhat-cloud-services/frontend-components/components/esm/Inventory';
import { act } from 'react-dom/test-utils';

const customMiddleWare = store => next => action => {
    next(action);
}

const mockStore = configureStore([customMiddleWare]);
let storeObj = { 
    SystemCvesStore: {
        cveList: {
            payload: {
                errors: undefined
            }
        }
    },
    entityDetails: {
        entity: 'testEntity',
        opt_out: false
    }
};

const props = {
    optOutSystem: jest.fn(() => Promise.resolve('hello world')),
    fetchSystemDetails: jest.fn(),
    addNotification: jest.fn(),
    match: { params: {inventoryId: 'testCve'} }
};

window.insights = { 
    loadInventory: (args) => new Promise((resolve, reject) => {
        return resolve({
            mergeWithEntities: () => {},
            mergeWithDetail: () => {},
            INVENTORY_ACTION_TYPES: ''
        })
    }) 
};

describe('SystemDetailPage', () => {
    it('Should match the snapshots', async () => {
        const tempStore = mockStore(storeObj);
        const wrapper = mountWithIntl(
            <Provider store={tempStore}>
                <Router>
                    <SystemDetailsPage { ...props } />
                </Router>
            </Provider>);
        await Promise.resolve();
        wrapper.update();
        expect(toJson(wrapper)).toMatchSnapshot();
    });


    it('Should call optOutSystem with true when entity exists and isOptOut is true', async () => {
        const tempStore = mockStore(storeObj);
        let wrapper = mountWithIntl(
            <Provider store={tempStore}>
                <Router>
                    <SystemDetailsPage.WrappedComponent {...props} systemDetails = {{opt_out: true, entity: true}} intl = {{ formatMessage: jest.fn()}}/>);
                </Router>
            </Provider>
        );
        await Promise.resolve();
        wrapper.update();            
        console.log(wrapper.find('InventoryDetail').instance().optOutSystem(false))
        expect(props.optOutSystem).toHaveBeenCalledWith('testCve', false);
        expect(props.fetchSystemDetails).toHaveBeenCalledWith('testCve');
    });

    it('Should call optOutSystem with false when entity exists and isOptOut is false', async () => {
        const tempStore = mockStore(storeObj);
        let wrapper = mountWithIntl(
            <Provider store={tempStore}>
                <Router>
                    <SystemDetailsPage.WrappedComponent {...props} systemDetails = {{opt_out: false, entity: true}} intl = {{ formatMessage: jest.fn()}}/>);
                </Router>
            </Provider>);
        await Promise.resolve();
        wrapper.update();
        console.log(wrapper.find('InventoryDetail').instance().optOutSystem(true))
        expect(props.optOutSystem).toHaveBeenCalledWith('testCve', true);
        expect(props.fetchSystemDetails).toHaveBeenCalledWith('testCve');
    });
});

