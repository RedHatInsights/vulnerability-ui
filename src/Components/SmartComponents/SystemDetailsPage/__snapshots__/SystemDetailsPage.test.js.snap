// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemDetailPage Should match the snapshots 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(injectIntl(Connect(InventoryDetail)))
        addNotification={[MockFunction]}
        fetchSystemDetails={[MockFunction]}
        match={
          Object {
            "params": Object {
              "inventoryId": "testCve",
            },
          }
        }
        optOutSystem={[MockFunction]}
      >
        <injectIntl(Connect(InventoryDetail))
          addNotification={[MockFunction]}
          fetchSystemDetails={[MockFunction]}
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
          optOutSystem={[MockFunction]}
        >
          <Connect(InventoryDetail)
            addNotification={[MockFunction]}
            fetchSystemDetails={[MockFunction]}
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            intl={
              Object {
                "defaultFormats": Object {},
                "defaultLocale": "en",
                "formatDate": [Function],
                "formatDateToParts": [Function],
                "formatDisplayName": [Function],
                "formatList": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatNumberToParts": [Function],
                "formatPlural": [Function],
                "formatRelativeTime": [Function],
                "formatTime": [Function],
                "formatTimeToParts": [Function],
                "formats": Object {},
                "formatters": Object {
                  "getDateTimeFormat": [Function],
                  "getDisplayNames": [Function],
                  "getListFormat": [Function],
                  "getMessageFormat": [Function],
                  "getNumberFormat": [Function],
                  "getPluralRules": [Function],
                  "getRelativeTimeFormat": [Function],
                },
                "locale": "en",
                "messages": Object {
                  "default.cancel": "Cancel",
                  "default.delete": "Delete",
                  "default.remove": "Remove",
                  "default.save": "Save",
                },
                "onError": [Function],
                "textComponent": Symbol(react.fragment),
                "timeZone": undefined,
                "wrapRichTextChunksInFragment": undefined,
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            optOutSystem={[MockFunction]}
          >
            <InventoryDetail
              addNotification={[Function]}
              fetchSystemDetails={[Function]}
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              intl={
                Object {
                  "defaultFormats": Object {},
                  "defaultLocale": "en",
                  "formatDate": [Function],
                  "formatDateToParts": [Function],
                  "formatDisplayName": [Function],
                  "formatList": [Function],
                  "formatMessage": [Function],
                  "formatNumber": [Function],
                  "formatNumberToParts": [Function],
                  "formatPlural": [Function],
                  "formatRelativeTime": [Function],
                  "formatTime": [Function],
                  "formatTimeToParts": [Function],
                  "formats": Object {},
                  "formatters": Object {
                    "getDateTimeFormat": [Function],
                    "getDisplayNames": [Function],
                    "getListFormat": [Function],
                    "getMessageFormat": [Function],
                    "getNumberFormat": [Function],
                    "getPluralRules": [Function],
                    "getRelativeTimeFormat": [Function],
                  },
                  "locale": "en",
                  "messages": Object {
                    "default.cancel": "Cancel",
                    "default.delete": "Delete",
                    "default.remove": "Remove",
                    "default.save": "Save",
                  },
                  "onError": [Function],
                  "textComponent": Symbol(react.fragment),
                  "timeZone": undefined,
                  "wrapRichTextChunksInFragment": undefined,
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              optOutSystem={[Function]}
              systemDetails={
                Object {
                  "entity": "testEntity",
                  "opt_out": false,
                }
              }
            >
              <Page
                defaultManagedSidebarIsOpen={true}
                isManagedSidebar={false}
                mainTabIndex={-1}
                onPageResize={[Function]}
              >
                <div
                  className="pf-c-page"
                >
                  <main
                    className="pf-c-page__main"
                    tabIndex={-1}
                  >
                    <withRouter(Connect(VulnerabilityHeader))
                      title=""
                    >
                      <Connect(VulnerabilityHeader)
                        history={
                          Object {
                            "action": "POP",
                            "block": [Function],
                            "createHref": [Function],
                            "go": [Function],
                            "goBack": [Function],
                            "goForward": [Function],
                            "length": 1,
                            "listen": [Function],
                            "location": Object {
                              "hash": "",
                              "pathname": "/",
                              "search": "",
                              "state": undefined,
                            },
                            "push": [Function],
                            "replace": [Function],
                          }
                        }
                        location={
                          Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          }
                        }
                        match={
                          Object {
                            "isExact": true,
                            "params": Object {},
                            "path": "/",
                            "url": "/",
                          }
                        }
                        title=""
                      >
                        <VulnerabilityHeader
                          dispatch={[Function]}
                          history={
                            Object {
                              "action": "POP",
                              "block": [Function],
                              "createHref": [Function],
                              "go": [Function],
                              "goBack": [Function],
                              "goForward": [Function],
                              "length": 1,
                              "listen": [Function],
                              "location": Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "",
                                "state": undefined,
                              },
                              "push": [Function],
                              "replace": [Function],
                            }
                          }
                          location={
                            Object {
                              "hash": "",
                              "pathname": "/",
                              "search": "",
                              "state": undefined,
                            }
                          }
                          match={
                            Object {
                              "isExact": true,
                              "params": Object {},
                              "path": "/",
                              "url": "/",
                            }
                          }
                          showBreadcrumb={true}
                          title=""
                        >
                          <PageHeader>
                            <section
                              className="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
                              widget-type="InsightsPageHeader"
                            >
                              <PageHeaderTitle
                                title=""
                              >
                                <Component
                                  className=""
                                  headingLevel="h1"
                                  size="2xl"
                                  widget-type="InsightsPageHeaderTitle"
                                >
                                  <h1
                                    className="pf-c-title pf-m-2xl"
                                    widget-type="InsightsPageHeaderTitle"
                                  >
                                     
                                     
                                  </h1>
                                </Component>
                              </PageHeaderTitle>
                              <InventoryDetailHead
                                actions={
                                  Array [
                                    Object {
                                      "onClick": [Function],
                                      "title": "Exclude from vulnerability analysis",
                                    },
                                  ]
                                }
                                hideBack={true}
                              >
                                <div>
                                  A mock with '
                                  ' passed!
                                </div>
                              </InventoryDetailHead>
                            </section>
                          </PageHeader>
                        </VulnerabilityHeader>
                      </Connect(VulnerabilityHeader)>
                    </withRouter(Connect(VulnerabilityHeader))>
                    <Connect(Main)>
                      <Main>
                        <section
                          className="pf-l-page__main-section pf-c-page__main-section"
                          page-type=""
                        >
                          <AppInfo
                            optOutSystemHandler={[Function]}
                          >
                            <div>
                              this is inventory body
                            </div>
                          </AppInfo>
                        </section>
                      </Main>
                    </Connect(Main)>
                  </main>
                </div>
              </Page>
            </InventoryDetail>
          </Connect(InventoryDetail)>
        </injectIntl(Connect(InventoryDetail))>
      </withRouter(injectIntl(Connect(InventoryDetail)))>
    </Router>
  </BrowserRouter>
</Provider>
`;
